.circleci/config.yml

```diff
@@ -1,5 +1,166 @@
-version: 2
+version: 2.1
```

Comment:
I recall discussing switching to using 2.0 so we can run it in internal CircleCI instance as well, but use the circleci to generate `.circleci/config.yml` from 2.1 source.

Reply:
Not gonna do that in this PR bc we already don't use this config in enterprise and I wanna keep iterating on it. We can follow up with it when we want to use circle on ent.